span_compressed.go

```diff
@@ -264,13 +263,21 @@ func (s *SpanData) setCompressedSpanName() {
 	if s.composite.compressionStrategy != compressedStrategySameKind {
 		return
 	}
-	service := s.Context.destinationService.Resource
+	s.Name = s.getCompositeSpanName()
+}
 
-	var builder strings.Builder
-	builder.Grow(len(compressedSpanSameKindName) + len(service))
-	builder.WriteString(compressedSpanSameKindName)
-	builder.WriteString(service)
-	s.Name = builder.String()
+func (s *SpanData) getCompositeSpanName() string {
+	if s.Context.serviceTarget.Type == "" {
+		if s.Context.serviceTarget.Name == "" {
+			return compressedSpanSameKindName + "unknown"
+		} else {
+			return compressedSpanSameKindName + s.Context.serviceTarget.Name
```

Comment:
IIANM it's impossible to reach this code path, since we're guaranteed to infer a `service.target.type` value if `service.target.name` is set. Is that correct? Perhaps the spec should be updated to remove the pseudo-code path?

Reply:
This is correct. Should we merge this and open a followup PR once the spec is updated ? 

We could also remove the line since it's never reached. Let me know what you think!

Reply:
I think it's fine to leave it and follow up.